hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/hip904/TokenClaimAirdropTest.java

```diff
@@ -117,6 +138,162 @@ final Stream<DynamicTest> claimFungibleTokenAirdrop() {
+                                pendingAirdrop(OWNER, RECEIVER, FUNGIBLE_TOKEN),
+                                pendingAirdrop(OWNER, RECEIVER, FUNGIBLE_TOKEN),
+                                pendingAirdrop(OWNER_2, RECEIVER, FUNGIBLE_TOKEN_2),
+                                pendingNFTAirdrop(OWNER, RECEIVER, NON_FUNGIBLE_TOKEN, 1))
+                        .payingWith(RECEIVER)
+                        .feeUsd(0.001)
+                        .via("claimTxn")
+                        .hasPrecheck(PENDING_NFT_AIRDROP_ALREADY_EXISTS));
+    }
+
+    @HapiTest
+    @DisplayName("token claim with more than ten pending airdrops should fail")
+    final Stream<DynamicTest> tokenClaimWithMoreThanTenPendingAirdropsShouldFail() {
+        return hapiTest(
+                cryptoCreate(OWNER).balance(ONE_HUNDRED_HBARS),
+                cryptoCreate(RECEIVER).balance(ONE_HUNDRED_HBARS).maxAutomaticTokenAssociations(0),
+                tokenCreate(FUNGIBLE_TOKEN)
+                        .treasury(OWNER)
+                        .tokenType(FUNGIBLE_COMMON)
+                        .initialSupply(1000L),
+                tokenCreate(FUNGIBLE_TOKEN_2)
+                        .treasury(OWNER)
+                        .tokenType(FUNGIBLE_COMMON)
+                        .initialSupply(1000L),
+                tokenCreate(FUNGIBLE_TOKEN_3)
+                        .treasury(OWNER)
+                        .tokenType(FUNGIBLE_COMMON)
+                        .initialSupply(1000L),
+                tokenCreate(FUNGIBLE_TOKEN_4)
+                        .treasury(OWNER)
+                        .tokenType(FUNGIBLE_COMMON)
+                        .initialSupply(1000L),
+                tokenCreate(FUNGIBLE_TOKEN_5)
+                        .treasury(OWNER)
+                        .tokenType(FUNGIBLE_COMMON)
+                        .initialSupply(1000L),
+                tokenCreate(FUNGIBLE_TOKEN_6)
+                        .treasury(OWNER)
+                        .tokenType(FUNGIBLE_COMMON)
+                        .initialSupply(1000L),
+                tokenCreate(FUNGIBLE_TOKEN_7)
+                        .treasury(OWNER)
+                        .tokenType(FUNGIBLE_COMMON)
+                        .initialSupply(1000L),
+                tokenCreate(FUNGIBLE_TOKEN_8)
+                        .treasury(OWNER)
+                        .tokenType(FUNGIBLE_COMMON)
+                        .initialSupply(1000L),
+                tokenCreate(FUNGIBLE_TOKEN_9)
+                        .treasury(OWNER)
+                        .tokenType(FUNGIBLE_COMMON)
+                        .initialSupply(1000L),
+                tokenCreate(FUNGIBLE_TOKEN_10)
+                        .treasury(OWNER)
+                        .tokenType(FUNGIBLE_COMMON)
+                        .initialSupply(1000L),
+                tokenCreate(FUNGIBLE_TOKEN_11)
+                        .treasury(OWNER)
+                        .tokenType(FUNGIBLE_COMMON)
```

Comment:
you can create a helper method that creates a token and/or moves a token to reduce code duplication